enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestionResult/ResultTitle.tsx

```diff
@@ -0,0 +1,74 @@
+import { type ReactNode, isValidElement } from "react";
+
+import { Anchor, Stack, Text } from "metabase/ui";
+
+import { useInteractiveQuestionContext } from "../InteractiveQuestion/context";
+import { getQuestionTitle } from "../QuestionTitle";
+
+import type { InteractiveQuestionResultProps } from "./InteractiveQuestionResult";
+
+interface ResultTitleTextProps
+  extends Pick<InteractiveQuestionResultProps, "withResetButton"> {
+  text: string | ReactNode;
+  isQuestionChanged?: boolean;
+  onReset?: () => void;
+  originalName?: string | null;
+}
+
+const ResultTitleText = ({
+  text,
+  withResetButton = false,
+  isQuestionChanged = false,
+  onReset,
+  originalName,
+}: ResultTitleTextProps) => (
+  <Stack spacing="xs">
+    {originalName && withResetButton && isQuestionChanged && (
+      <Text fw={600} size="sm">
+        Return to
+        <Anchor size="sm" ml="xs" color="brand" onClick={onReset}>
+          {originalName}
+        </Anchor>
+      </Text>
+    )}
+    <Text fw={700} c="var(--mb-color-text-primary)" fz="xl">
+      {text}
+    </Text>
+  </Stack>
+);
+
+export const ResultTitle = ({
+  customTitle,
+  withTitle = true,
+  withResetButton = false,
+}: InteractiveQuestionResultProps) => {
+  const { question, originalQuestion, onReset } =
+    useInteractiveQuestionContext();
+
+  if (!withTitle) {
+    return null;
+  }
+
+  if (isValidElement(customTitle)) {
+    return customTitle;
+  }
+
+  const isQuestionChanged = originalQuestion
+    ? question?.isQueryDirtyComparedTo(originalQuestion)
+    : true;
+
+  const originalName = originalQuestion?.displayName();
+
+  const titleText =
+    !customTitle && question ? getQuestionTitle({ question }) : customTitle;
```

Comment:
Just to confirm, if `customTitle` is nullish and `question` is nullish, I think that would result in `customTitle` being `null` which would cause the title text to be empty.

Reply:
Ah, that's very true. let me change that.

Reply:
ok, i've modified it so that if customTitle is a string, then we still display that, otherwise we default to the value of getQuestionTitle

Reply:
Neato, thanks!